app-sre/qontract-reconcile
# Add integration terraform_cloudflare_users to manage cloudflare user access

Add new integration `terraform-cloudflare-users` for managing user management for cloudflare. 

Ticket: APPSRE-6865

Design Doc: https://gitlab.cee.redhat.com/service/app-interface/-/blob/master/docs/app-sre/design-docs/terraform-cloudflare-users-support.md

Schema dependency: https://github.com/app-sre/qontract-schemas/pull/379

Validation:

1. Assign a role to a user => cloudflare invite is sent and upon accepting the invite, the role is assigned to the user.

1. Remove a role from a user => If invite is sent but not yet accepted, invitation is cancelled. Otherwise if user is already in the account, then the user is removed.

1.  Repeat step 1 and 2 with multiple roles with different account combination and verify the result. 

The unit test also covers fair amount of possibilities for roles and account combination.
